gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -0,0 +1,480 @@
+
+  @Inject LocalDataStore localDataStore;
+
+  private static void assertEqualsIgnoreId(
+      ObservationMutation expected, ObservationMutation actual) {
+    // TODO: Id is auto-assigned to ObservationMutation.
+    //  If we try to give it while inserting, then it causes problems. Improve this behavior.
+    //  So, copy the id from actual to expected and then compare the objects.
+    expected = expected.toBuilder().setId(actual.getId()).build();
+    assertThat(expected).isEqualTo(actual);
+  }
+
+  private static void assertObservation(ObservationMutation mutation, Observation observation) {
+    assertThat(mutation.getObservationId()).isEqualTo(observation.getId());
+    assertThat(mutation.getFeatureId()).isEqualTo(observation.getFeature().getId());
+    assertThat(TEST_USER).isEqualTo(observation.getCreated().getUser());
+    assertThat(TEST_FORM).isEqualTo(observation.getForm());
+    assertThat(TEST_PROJECT).isEqualTo(observation.getProject());
+    assertThat(TEST_USER).isEqualTo(observation.getLastModified().getUser());
+    MatcherAssert.assertThat(
+        ResponseMap.builder().applyDeltas(mutation.getResponseDeltas()).build(),
+        samePropertyValuesAs(observation.getResponses()));
+  }
+
+  private static void assertFeature(String featureId, Point point, Feature feature) {
+    assertThat(featureId).isEqualTo(feature.getId());
+    assertThat(TEST_PROJECT).isEqualTo(feature.getProject());
+    assertThat(TEST_LAYER.getItemLabel()).isEqualTo(feature.getTitle());
+    assertThat(TEST_LAYER).isEqualTo(feature.getLayer());
+    assertThat(feature.getCustomId()).isNull();
+    assertThat(feature.getCaption()).isNull();
+    assertThat(point).isEqualTo(feature.getPoint());
+    assertThat(TEST_USER).isEqualTo(feature.getCreated().getUser());
+    assertThat(TEST_USER).isEqualTo(feature.getLastModified().getUser());
+  }
+
+  @Before
+  public void setUp() {
+    DaggerTestComponent.create().inject(this);
+  }
+
+  @Test
+  public void testInsertProject() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).test().assertComplete();
+  }
+
+  @Test
+  public void testGetProjects() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.getProjects().test().assertValue(ImmutableList.of(TEST_PROJECT));
+  }
+
+  @Test
+  public void testGetProjectById() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.getProjectById("project id").test().assertValue(TEST_PROJECT);
+  }
+
+  @Test
+  public void testDeleteProject() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.deleteProject(TEST_PROJECT).test().assertComplete();
+    localDataStore.getProjects().test().assertValue(AbstractCollection::isEmpty);
+  }
+
+  @Test
+  public void testInsertUser() {
+    localDataStore.insertOrUpdateUser(TEST_USER).test().assertComplete();
```

Comment:
In `testDeleteProject` we checked that the project was actually deleted. Should we do the same here for user insertion? The method could then be called `testInsertAndGetUser` and the next method deleted.